backend/src/test/java/ca/bc/gov/backendstartapi/repository/FavouriteActivityRepositoryTest.java

```diff
@@ -0,0 +1,128 @@
+import ca.bc.gov.backendstartapi.enums.ActivityEnum;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = Replace.NONE)
+@TestMethodOrder(OrderAnnotation.class)
+class FavouriteActivityRepositoryTest {
+
+  @Autowired private FavouriteActivityRepository favouriteActivityRepository;
+
+  @Autowired private UserProfileRepository userProfileRepository;
+
+  private static final String USER_ID = "123456789123456789@idir";
+
+  @Test
+  @DisplayName("createWithDefaultValuesTest")
+  @Order(1)
+  void createWithDefaultValuesTest() {
+    FavouriteActivityEntity activity = new FavouriteActivityEntity();
+    activity.setUserId(USER_ID);
+    activity.setActivity(ActivityEnum.SEEDLOT_REGISTRATION);
+    FavouriteActivityEntity created = favouriteActivityRepository.save(activity);
+
+    Assertions.assertEquals(1L, created.getId());
+    Assertions.assertEquals(USER_ID, created.getUserId());
+    Assertions.assertEquals(ActivityEnum.SEEDLOT_REGISTRATION, created.getActivity());
+    Assertions.assertFalse(created.getHighlighted());
+    Assertions.assertTrue(created.getEnabled());
+  }
+
+  @Test
+  @DisplayName("createWithAllValuesTest")
+  @Order(2)
+  void createWithAllValuesTest() {
+    FavouriteActivityEntity activity = new FavouriteActivityEntity();
+    activity.setUserId(USER_ID);
+    activity.setActivity(ActivityEnum.SEEDLING_REQUEST);
+    activity.setHighlighted(true);
+    activity.setEnabled(false);
+    FavouriteActivityEntity created = favouriteActivityRepository.save(activity);
+
+    Assertions.assertEquals(ActivityEnum.SEEDLING_REQUEST, created.getActivity());
+    Assertions.assertEquals(USER_ID, created.getUserId());
+    Assertions.assertTrue(created.getHighlighted());
+    Assertions.assertFalse(created.getEnabled());
+  }
+
+  @Test
+  @DisplayName("preventFromDuplicateUserTest")
+  @Order(3)
+  void preventFromDuplicateUserTest() {
+    FavouriteActivityEntity activity = new FavouriteActivityEntity();
+    activity.setUserId(USER_ID);
+    activity.setActivity(ActivityEnum.SEEDLOT_REGISTRATION);
+    favouriteActivityRepository.save(activity);
+  }
```

Comment:
It feels like we should have some assertion here, but I checked the code and it seems like nothing will happen because the duplicate prevention logic is in `FavouriteActivityService`.

Thinking about it now, an unique constraint in the `favourite_activity` table for the pair `user_id, activity` would give us what we want, wouldn't it? Then we could remove the search we do in `FavouriteActivityService.createUserActivity`.

Reply:
Yeah, I think I can remove this test case, since the possibility of duplication is been covered on the service class. 

Related to the unique constraint, I agree, indeed we could have it on the database, but since we already have it on the code site, I think we're good to go.